rawx-lib/src/attr_handler.c

```diff
@@ -168,6 +194,64 @@ _get (int fd, const char *k, gchar **pv)
 
 #define GET(K,R) _get(fd, ATTR_DOMAIN "." K, &(R))
 
+
+/**
+ * Fullpaths are in the form 'oio:account/container/path'
+ * Sometimes we need to only have the fullpath of a specific path
+ * So we have to check the prefix:'oio:' and the suffix '/path' to
+ * find the corresponding fullpath
+ */
+static GString*
+_get_fullpaths_from_listxattr(gchar * xattrlist, ssize_t size, gchar *path)
+{
+	GString* user_list = g_string_new("");
+	gchar* elem = NULL;
+	gchar* suffix = g_strconcat("/", path, NULL);
+	gint suffix_length = strlen(suffix);
+	for (gint i = 0; i < size; i+= strlen(&xattrlist[i]) + 1) {
+		elem = strstr(&xattrlist[i], ATTR_DOMAIN_OIO);
+		if (elem) {
+			if (path)
+				elem = g_strrstr(&xattrlist[i], suffix);
+			if(!path || (elem + suffix_length) == (xattrlist + i + suffix_length)) {
```

Comment:
There is something wrong around here. `(elem + suffix_length) == (xattrlist + i + suffix_length)` is the same as `elem == (xattrlist + i)`.